src/tribler-core/tribler_core/start_core.py

```diff
@@ -174,12 +174,15 @@ def start_tribler_core(base_path, api_port, api_key, root_state_dir, gui_test_mo
     loop = asyncio.get_event_loop()
     loop.set_exception_handler(CoreExceptionHandler.unhandled_error_observer)
 
-    loop.run_until_complete(core_session(config, components=list(components_gen(config))))
-
-    if trace_logger:
-        trace_logger.close()
-
-    process_checker.remove_lock_file()
```

Comment:
We must always remove the lock file on Tribler crash to re-run the application tester reliably, so try/finally block is necessary here.

Reply:
Our lock file mechanics are broken, as it was built on the assumption that OS does not reuse PIDs before system restart. This is not true, Windows especially reuses process IDs quickly, and that leads to many non-obvious bugs. I will open a separate issue for that.